timesketch/frontend-ng/src/views/Story.vue

```diff
@@ -0,0 +1,373 @@
+                        </v-list-item-content>
+                      </v-list-item>
+                    </v-list-item-group>
+                  </v-list>
+                </v-card>
+              </v-menu>
+            </div>
+          </div>
+        </v-hover>
+      </div>
+    </div>
+  </v-container>
+</template>
+
+<script>
+import ApiClient from '../utils/RestApiClient'
+import EventBus from '../main'
+import DOMPurify from 'dompurify'
+import { marked } from 'marked'
+import _ from 'lodash'
+
+import TsEventList from '../components/Explore/EventList'
+
+const defaultBlock = () => {
+  return {
+    componentName: '',
+    componentProps: {},
+    content: '',
+    draft: '',
+    edit: true,
+    isActive: false,
+  }
+}
+
+const componentCompatibility = () => {
+  return {
+    TsViewEventList: 'TsEventList',
+  }
+}
+
+export default {
+  props: ['sketchId', 'storyId'],
+  components: { TsEventList },
+  data: function () {
+    return {
+      title: '',
+      titleDraft: '',
+      blocks: [],
+      renameStoryDialog: false,
+    }
+  },
+  computed: {
+    sketch() {
+      return this.$store.state.sketch
+    },
+    meta() {
+      return this.$store.state.meta
+    },
+    hasContent() {
+      if (!this.blocks.length) {
+        return false
+      }
+      if (this.blocks.length < 2 && !this.blocks[0].componentName && !this.blocks[0].content) {
+        return false
+      }
+      return true
+    },
+  },
+  methods: {
+    updateDraft: _.debounce(function (e, block) {
+      block.draft = e
+    }, 300),
+    fetchStory() {
+      ApiClient.getStory(this.sketchId, this.storyId)
+        .then((response) => {
+          this.title = response.data.objects[0].title
+          this.titleDraft = this.title
+          let content = response.data.objects[0].content
+          if (content === '[]') {
+            this.blocks = [defaultBlock()]
+          } else {
+            this.blocks = JSON.parse(content)
+          }
+          this.formatBlocks(this.blocks)
+          // Set first page in edit mode if the story is empty
+          if (this.blocks.length < 2 && !this.blocks[0].componentName && !this.blocks[0].content) {
+            this.blocks[0].edit = true
+          }
+        })
+        .catch((e) => {
+          console.error(e)
+        })
+    },
+    formatBlocks(blocks) {
+      // Format block to support backwards compatibility for old style blocks
+      let compat = componentCompatibility()
+      blocks.forEach((block) => {
+        if (block.componentName in compat) {
+          block.componentName = compat[block.componentName]
+        }
+      })
+      return blocks
+    },
+    formatComponentProps(block) {
+      // Backwards compatibility for old style TsViewEventList props
+      if (block.componentName === 'TsEventList' || block.componentName === 'TsViewEventList') {
+        const EVENTS_PER_PAGE = 10
+        let queryFilter = JSON.parse(block.componentProps.view.filter)
```

Comment:
As discussed offline: `block.componentProps.view.filter` can be undefined and hence preventing the render of the `TsEventList` block. Please add a fix and then we are good to go.

Reply:
fixed